avocado/core/dispatcher.py

```diff
@@ -59,3 +59,26 @@ class CLICmdDispatcher(Dispatcher):
 
     def __init__(self):
         super(CLICmdDispatcher, self).__init__('avocado.plugins.cli.cmd')
+
+
+class JobPrePostDispatcher(Dispatcher):
+
+    """
+    Calls extensions before Job execution
+
+    Automatically adds all the extension with entry points registered under
+    'avocado.plugins.job.prepost'
+    """
+
+    def __init__(self):
+        super(JobPrePostDispatcher, self).__init__('avocado.plugins.job.prepost')
+
+    def map_methods(self, method_name, job):
+        for ext in self.extensions:
+            try:
+                if hasattr(ext.obj, method_name):
+                    method = getattr(ext.obj, method_name)
+                    method(job)
+            except Exception as e:
+                job.log.error('Error running method "%s" of plugin "%s": %s',
+                              method_name, ext.name, e)
```

Comment:
Yep, it works. It does not affect the exit code, but the same applies to plugin initialization, so I think we should keep it this way.

But for the future, we should consider adding "PLUGIN_ERROR" exit code.